### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by muydanny**
Fixed application layout so our welcome page doesnt show up on every page. Following factory bot. User dashboard with favorites.

*Updated w/ group to include previous Oauth PR

### CODE REVIEW

File: app/views/welcome/index.html.erb [hunk 1]
```diff
@@ -1,6 +1,46 @@
-<h1> Podsmack </h1>
+<header class="masthead row">
```
Comment (muydanny) [inline app/views/welcome/index.html.erb:1]: This was moved from application. So now just our welcome page has this.

File: spec/features/user/logged_in_user_dashboard.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+describe 'User logged in viewing dashboard' do
+  it 'A logged in user visits /dashboard and sees dashboard attributes' do
+    @user = create(:user, password: 'password')
+
+    visit '/'
```
Comment (muydanny) [inline spec/features/user/logged_in_user_dashboard.rb]: Taken out later. Waiting for Oauth completion

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,8 @@
 class ApplicationController < ActionController::Base
+  helper_method :current_user
+
```
Comment (muydanny) [inline app/controllers/application_controller.rb:3]: Created for spec to create instance of user.

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1,11 +1,18 @@
 <h2> Welcome, <%= @user.user_name %> </h2>
 
+<h3> Following </h3>
```
Comment (muydanny) [inline app/views/users/show.html.erb:3]: Need to distinguish between favorite and following.

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
-      <h1><%= @user.favorite_podcasts.length %>Favorites</h1>
-      <% @user.favorite_podcasts.each do |podcast| %>
-        <section id="follow">
-            <%= podcast.name %>
-        </section>
-      <% end %>
+  <h1><%= @user.favorite_podcasts.length %>Favorites</h1>
+  <% @user.favorite_podcasts.each do |podcast| %>
+    <section id="follow">
+      <%= podcast.name %>
```
Comment (muydanny) [inline app/views/users/show.html.erb:8]: Add some sort of styling like Bulleted or listed alphabetically. (maybe)

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -1,11 +1,18 @@
...
     </section>
-  </section>
\ No newline at end of file
+  <% end %>
+</section>
+
+
+<h3> Edit Profile </h3>
+
+<h4> Podcast Application </h4>
```
Comment (muydanny) [inline app/views/users/show.html.erb:16]: Change to 'Upload Podcast' || 'Want to upload your podcast?'

File: spec/factories/followings.rb [hunk 7]
```diff
@@ -1,6 +1,7 @@
 FactoryBot.define do
   factory :following do
-    podcast { nil }
-    user { nil }
+    association :podcast, factory: :podcast
```
Comment (janegreene) [inline spec/factories/followings.rb:3]: This is a cool trick, thanks Danny.

File: spec/features/user/logged_in_user_dashboard.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+describe 'User logged in viewing dashboard' do
+  it 'A logged in user visits /dashboard and sees dashboard attributes' do
+    @user = create(:user, password: 'password')
```
Comment (janegreene) [inline spec/features/user/logged_in_user_dashboard.rb:5]: We will no longer have passwords on our end.

File: app/models/user.rb [hunk 9]
```diff
@@ -6,6 +6,11 @@ class User < ApplicationRecord
   validates :user_name, uniqueness: true, presence: true
```
Comment (takeller) [inline app/models/user.rb:6]: user_name no longer needs to be unique. We will enforce uniqueness on the uid as given by google oauth

**Pull request merged**